pages/api/stripe_webhook.ts

```diff
@@ -169,6 +169,11 @@ const webhook = async (req: NextApiRequest, res: NextApiResponse) => {
         },
       })
       await supportingSignup(event.data.object.customer_email)
+      // If it's a sponsored supporter, make sure to cancel the subscription immediately
+      // Sponsored memberships last only for a year
+      if (event.data.object.lines.data[0].metadata.subType === "sponsored") {
```

Comment:
A very tiny remark: this line does a lot of digging into property. I wondered if it would be better to abstract a bit for readability. For example, we could construct `isSponsored` variable to use in the if condition to improve readability for the if condition. Just a thought 👋